modules/v1/workspaces/resources/TimelinePostResource.php

```diff
@@ -0,0 +1,143 @@
+            'file_url' => function () {
+                return $this->getFileUrl();
+            },
+            'created_at' => function () {
+                return $this->created_at * 1000;
+            },
+            'updated_at' => function () {
+                return $this->updated_at * 1000;
+            },
+        ];
+    }
+
+    public function attributes()
+    {
+        return ArrayHelper::merge(array_keys(parent::attributeLabels()), ['workspace_id']);
+    }
+
+    public function rules()
+    {
+        return array_merge(parent::rules(), [[['workspace_id'], 'integer']]);
+    }
+
+    public function extraFields()
+    {
+        return ['workspace', 'workspaceTimelinePosts', 'workspace_id', 'createdBy'];
+    }
+
+    /**
+     * @return ActiveQuery
+     */
+    public function getWorkspace()
+    {
+        return $this->hasOne(WorkspaceResource::class, ['id' => 'timeline_post_id'])
+            ->via('workspaceTimelinePosts');
+    }
+
+    /**
+     * @return WorkspaceTimelinePostQuery|ActiveQuery
+     */
+    public function getWorkspaceTimelinePosts()
+    {
+        return $this->hasMany(WorkspaceTimelinePostResource::class, ['timeline_post_id' => 'id']);
+    }
+
+    /**
+     * After save Timeline Post create new Workspace Timeline Post
+     *
+     * @param $insert
+     * @param $changedAttributes
+     * @throws ValidationException
+     */
+    public function afterSave($insert, $changedAttributes)
+    {
+        parent::afterSave($insert, $changedAttributes);
+        if ($insert) {
+            $workspaceTimelinePosts = new WorkspaceTimelinePostResource();
+            $workspaceTimelinePosts->workspace_id = $this->workspace_id;
+            $workspaceTimelinePosts->timeline_post_id = $this->id;
+            if (!$workspaceTimelinePosts->save()) {
+                throw new ValidationException(\Yii::t('app', 'Unable to create Workspace Timeline Post'));
+            }
+        }
+    }
+
+    /**
+     * Delete Workspace Timeline Posts before Timeline Post is deleted
+     */
+    public function beforeDelete()
+    {
+        WorkspaceTimelinePostResource::deleteAll(['timeline_post_id' => $this->id]);
+        return parent::beforeDelete();
+    }
+
+    public function load($data, $formName = null)
+    {
+        $this->file = UploadedFile::getInstanceByName('file');
+
+        return parent::load($data, $formName);
+    }
+
+    /**
+     * @param bool $runValidation
+     * @param null $attributeNames
+     * @return bool
+     * @throws Exception
+     */
+    public function save($runValidation = true, $attributeNames = null)
+    {
+        if (!$this->file) {
+            return parent::save($runValidation, $attributeNames);
+        }
+        $dirPath = '/timelinePosts/' . $this->workspace_id;
+        $this->file_path = $dirPath . '/' . Yii::$app->security->generateRandomString() . '.' . $this->file->extension;
+
+        $parentSave = parent::save($runValidation, $attributeNames);
```

Comment:
@skavt Please use the same logic as in other pull request. Move parent save to end.